### PR TITLE
Add Term Entry: JavaScript Strings: `.slice()` method

### LINKED ISSUE #588: [Term Entry] JavaScript Strings: `.slice()`
### Reason/inspiration (optional)

Please write a term entry about the `.slice()` method for **JavaScript strings** that includes:

* A "Syntax" section with a ```pseudo code snippet.
* An "Example" section with a brief example ```js code snippet.
* A "Codebyte Example" section with a ```codebyte/javascript snippet.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] I agree to follow Codecademy Doc's Code of Conduct

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by denisCabrera**
### Description

Created slice.md file for the  `.slice()`  JavaScript method. Included syntax section, an example and a codebyte.
Closes #588 
### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by denisCabrera** [comment]
@Dusch4593 @yangc95 @SSwiniarski  @KTom101 Hope everyone is having a great week. This entry took me a while to complete since I have been busy with school and getting ready for deployment. However, I will have access to the internet and continue to contribute as much as I can 😃. I am ready for some feedback 👍

**Comment by SSwiniarski** [comment]
@denisCabrera, this is actually a duplicate of this [entry](https://www.codecademy.com/resources/docs/javascript/arrays/slice), apparently someone forgot to close the issue. Apologies for that. @Dusch4593, @yangc95, @KTom101: anything from here that we can incorporate into the existing entry?

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
This should be edited to reflect the **string** `.slice()` method.

Also should be moved to javascript/concepts/strings/terms/slice/slice.md

**Comment by Dusch4593** [comment]
@denisCabrera I agree with @SSwiniarski on this one. We could refocus this entry on the JS String `.slice()` method. 😄

If you have any questions, feel free to let one of us know! Happy to help. 👍

File: content/javascript/concepts/strings/terms/slice/slice.md
Comment (Dusch4593) [inline content/javascript/concepts/strings/terms/slice/slice.md:22]:
```suggestion
string.slice(start);

string.slice(start, stop);
```

File: content/javascript/concepts/strings/terms/slice/slice.md
Comment (Dusch4593) [inline content/javascript/concepts/strings/terms/slice/slice.md:16]:
```suggestion
The `.slice()` method selects and returns a portion of a string.
```

File: content/javascript/concepts/strings/terms/slice/slice.md
Comment (Dusch4593) [inline content/javascript/concepts/strings/terms/slice/slice.md:26]:
```suggestion
- The `start` index is optional and begins at zero (inclusive).
- The `stop` index is also optional and defaults to the length of the string (not inclusive).
```

File: content/javascript/concepts/strings/terms/slice/slice.md
Comment (Dusch4593) [inline content/javascript/concepts/strings/terms/slice/slice.md:46]:
```suggestion
```codebyte/javascript
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@denisCabrera I went ahead and moved the entry to the correct position in the folder structure. I also started edits to replace all mentions of arrays with strings.

For my review, I left some comments/suggestions but wanted to also offer some high-level comments:

* Replace all mentions of arrays with strings and string terms.
* In the "Example" and "Codebyte Example" sections, rewrite so that strings are used instead of arrays.

**Comment by denisCabrera** [comment]
@Dusch4593  Thank you so much for taking the time to move this method into the correct file path . I have been traveling for the last few days and now getting some time to checkout the comments. I will like to take another look at the array section and see if I am missing anything else. You mentioned we should use strings, you could elaborate more on that ? 😃

**Comment by Dusch4593** [comment]
Hey @denisCabrera! 👋🏻 Just wanted to check in and see how everything is going on your side of things? I'm happy to help in any way I can 😄

**Comment by denisCabrera** [comment]
@Dusch4593  Hi, thank you for checking in. I have been getting settled here in the Philippines, getting used to the new time zone, work schedule and what not. I am finally getting a routine going and getting ready to finish some of the work I started on here. 😄

**Comment by Dusch4593** [comment]
> @Dusch4593 Hi, thank you for checking in. I have been getting settled here in the Philippines, getting used to the new time zone, work schedule and what not. I am finally getting a routine going and getting ready to finish some of the work I started on here. 😄

That's great to hear! Congratulations on the new move! 🎉 🎊 🎈 

I'm planning on moving as well, later this year. 😄

**Comment by denisCabrera** [comment]
@Dusch4593  I am actually here for only a few months. 😞

File: content/javascript/concepts/strings/terms/slice/slice.md
Comment (yangc95) [inline content/javascript/concepts/strings/terms/slice/slice.md:45]:
```suggestion
In the example below, the `animals` array is created. Then the `birds` array is created to extract the birds "eagles" and "owls". Lastly, the `birds` array is used to output a sentence.
```
```suggestion
In the example below the animals array is created. Then the birds array is created to extract the birds "eagles" and "owls". Lastly, the birds array is used to output a sentence.
```

Comment (yangc95) [inline content/javascript/concepts/strings/terms/slice/slice.md:45]:
```suggestion
In the example below, the `animals` array is created. Then the `birds` array is created to extract the birds "eagles" and "owls". Lastly, the `birds` array is used to output a sentence.
```
```suggestion
In the example below the animals array is created. Then the birds array is created to extract the birds "eagles" and "owls". Lastly, the birds array is used to output a sentence.
```

File: content/javascript/concepts/strings/terms/slice/slice.md
Comment (yangc95) [inline content/javascript/concepts/strings/terms/slice/slice.md:26]: Can we include what the default value will be if nothing is specified?

Comment (yangc95) [inline content/javascript/concepts/strings/terms/slice/slice.md:26]: Can we include what the default value will be if nothing is specified?

File: content/javascript/concepts/strings/terms/slice/slice.md
Comment (yangc95) [inline content/javascript/concepts/strings/terms/slice/slice.md:16]:
```suggestion
The `.slice()` method selects and returns a selected portion of a string.
```

Comment (yangc95) [inline content/javascript/concepts/strings/terms/slice/slice.md:16]:
```suggestion
The `.slice()` method selects and returns a selected portion of a string.
```

**Comment by yangc95** [comment]
@denisCabrera oh man, it must be tough having to move around so much, although it's nice getting to experience different places 🙂

I left a few comments with only minor suggestions. @Dusch4593 @SSwiniarski how is this entry looking on your end?

File: content/javascript/concepts/strings/terms/slice/slice.md
Comment (yangc95) [inline content/javascript/concepts/strings/terms/slice/slice.md:10]:
```suggestion
  - 'Strings'
```

File: content/javascript/concepts/strings/terms/slice/slice.md
Comment (yangc95) [inline content/javascript/concepts/strings/terms/slice/slice.md:40]:
```suggestion
let smoothie = 'mango strawberry pineapple';

let berry = smoothie.slice(6, 16);

console.log(fruit);
// Output: mango strawberry pineapple

console.log(berry);
// Output: strawberry
```

File: content/javascript/concepts/strings/terms/slice/slice.md
Comment (yangc95) [inline content/javascript/concepts/strings/terms/slice/slice.md:52]:
```suggestion
let animals = 'elephants, eagles, owls, and tigers';

console.log(`${animals.slice(11,17)} and ${animals.slice(19,23)} are cool birds`);
```

Comment (yangc95) [reply content/javascript/concepts/strings/terms/slice/slice.md:52]: @denisCabrera I made some code example suggestions, although feel free to adjust as desired!

File: content/javascript/concepts/strings/terms/slice/slice.md
Comment (yangc95) [reply content/javascript/concepts/strings/terms/slice/slice.md:10]: This is directly correlated to @Dusch4593 's comment ☺️

**Comment by denisCabrera** [comment]
@yangc95 @Dusch4593  Thank you for the feedback. This is why I love this place, I am always learning new things. All of you are super helpful. I am trying to get back into a routine of contributing as I set my new schedule here. School and work have been occupying a lot of my time lately. However, I am now picking up on things. Looking forward to more feedback and finishing all the commits I still got on here 😅

**Comment by denisCabrera** [comment]
@yangc95  I finished all the suggestions. Let me know how it looks. I added the default position for the start and end. Were you referring to nothing as in an empty method ?

File: content/javascript/concepts/strings/terms/slice/slice.md
Comment (Dusch4593) [inline content/javascript/concepts/strings/terms/slice/slice.md:32]:
```suggestion
let smoothie = "mango strawberry pineapple";
```

Let's use double-quotes for JS strings (I'm pretty sure other similar use them as well).

File: content/javascript/concepts/strings/terms/slice/slice.md
Comment (Dusch4593) [inline content/javascript/concepts/strings/terms/slice/slice.md:36]: Getting an error because we got rid of `fruit`. Let's make the following edit:
```suggestion
console.log(smoothie);
```

File: content/javascript/concepts/strings/terms/slice/slice.md
Comment (Dusch4593) [inline content/javascript/concepts/strings/terms/slice/slice.md:42]: For example snippets, let's include a short sentence that preview a \```shell snippet with the output:

```suggestion

The output will look like the following:

```

And then the snippet would look like this:

```shell
mango strawberry pineapple
strawberry
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@denisCabrera Thank you for the kind words! Always a pleasure to work with you 😄

Double thank you for the changes you just pushed. Much appreciated! I left a few additional comments/suggestions for you to check out. 😃

**Comment by yangc95** [comment]
> @yangc95 I finished all the suggestions. Let me know how it looks. I added the default position for the start and end. Were you referring to nothing as in an empty method ?

Yeah! It's perf now, I see you added the default value for `start` is 0 so everything is good on my end~

**Review by yangc95 — APPROVED** [review]

File: content/javascript/concepts/strings/terms/slice/slice.md
Comment (Dusch4593) [inline content/javascript/concepts/strings/terms/slice/slice.md:26]:
```suggestion
- The `start` index is optional and begins at zero (inclusive) by default.
```

File: content/javascript/concepts/strings/terms/slice/slice.md
Comment (Dusch4593) [inline content/javascript/concepts/strings/terms/slice/slice.md:27]:
```suggestion
- The `stop` index is also optional and defaults to the length of the string (not inclusive).
```

File: content/javascript/concepts/strings/terms/slice/slice.md
Comment (Dusch4593) [inline content/javascript/concepts/strings/terms/slice/slice.md:30]: Like in the "Codebyte Example", let's include 1 or 2 sentences here about what is going on in the example snippet below.

File: content/javascript/concepts/strings/terms/slice/slice.md
Comment (Dusch4593) [inline content/javascript/concepts/strings/terms/slice/slice.md:45]:
```suggestion
In the example below, an `animals` string is created. Next, the `.slice()` method is used to pick out substrings of certain animals, specifically birds:
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@denisCabrera This entry is looking great! 😄 I think after we've resolved these last few comments/suggestions we should be near ready for merge.

**Comment by yangc95** [comment]
> @yangc95 @Dusch4593 Hi, I finished all the suggested changes. I am currently working on running the scripts to fix the formatting tests that are failing and commit those changes. Thank you. I am a little rusty since my break 😅 . Let me know if there is anything else.

Sounds good! Let us know if you need assistance with these tests as well. @SSwiniarski @Dusch4593 how is everything looking on your end? Will merge once we get the approvals and these tests passing 👍

File: content/javascript/concepts/strings/terms/slice/slice.md
Comment (SSwiniarski) [inline content/javascript/concepts/strings/terms/slice/slice.md:39]: These comments are redundant with the output listed below. I'd remove them.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
I had one minor change.

File: content/javascript/concepts/strings/terms/slice/slice.md
Comment (denisCabrera) [reply content/javascript/concepts/strings/terms/slice/slice.md:39]: I agree. Thank you

**Comment by denisCabrera** [comment]
@yangc95 @Dusch4593 @SSwiniarski  I ran all the checks and fixed the latest suggestions thank you @SSwiniarski  good catch there at the end. I appreciate you all. I am convinced programming is some time of sorcery, after trying and trying I finally got test commands to work lol 😄 . Let me know if there is anything else. ✌️

**Review by SSwiniarski — APPROVED** [review]
LGTM!

**Review by Dusch4593 — APPROVED** [review]
LGTM! 👍🏻

**Comment by Dusch4593** [comment]
Congratulations @denisCabrera, the entry is now live! You can view it here: https://www.codecademy.com/resources/docs/javascript/strings/slice

Thank you so much for your persistence and hard work on this entry! 👏🏻 😄 🎉

**Pull request merged**
